course3/homework-string/test.cc

```diff
@@ -65,19 +66,29 @@ void string_test()
 	// operator +=
 	plusResult += string2;
 	assert(plusResult == "Hex123Hex");
+
+	// move contructor and assignment operator tests
+	const char* string_c_str_pointer = string3.c_str();
+	StringType string4 = std::move(string3);
+	assert(string_c_str_pointer == string4.c_str());
+
+	StringType string5;
+	string5 = std::move(string4);
+	assert(string_c_str_pointer == string5.c_str());
```

Comment:
The above unit tests pass for MyString, but not for std::string. This page explains why:
http://stackoverflow.com/questions/29806718/does-stdmove-on-stdstring-garantee-that-c-str-returns-same-result

I tried playing around with the unique_ptr idea mentioned in the first answer, but doing so just calls the move constructor (or assignment operator) on the unique_ptr, and not the underlying object, thus making the test moot.